ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1430,6 +1452,46 @@ impl Engine for AuthorityRound {
 		block_reward::apply_block_rewards(&rewards, block, &self.machine)
 	}
 
+	/// Make calls to the randomness contract.
+	fn on_prepare_block(&self, block: &ExecutedBlock) -> Result<Vec<SignedTransaction>, Error> {
+		let opt_signer = self.signer.read();
+		let signer = match opt_signer.as_ref() {
+			Some(signer) => signer,
+			None => return Ok(Vec::new()), // We are not a validator, so we shouldn't call the contracts.
+		};
+		let our_addr = signer.address();
+		let client = self.client.read().as_ref().and_then(|weak| weak.upgrade()).ok_or_else(|| {
+			debug!(target: "engine", "Unable to prepare block: missing client ref.");
+			EngineError::RequiresClient
+		})?;
+		let full_client = client.as_full_client()
+			.ok_or(EngineError::FailedSystemCall("Failed to upgrade to BlockchainClient.".to_string()))?;
+
+		// Our current account nonce. The transactions must have consecutive nonces, starting with this one.
```

Comment:
Seems this (obsolete) comment relates to implementation in our fork where we also performed other transactions (like calling `emitInitiateChange` function).

Reply:
I see, so maybe somewhat related to https://github.com/paritytech/parity-ethereum/pull/11245?

Reply:
Yes, here: https://github.com/paritytech/parity-ethereum/pull/11245/files#diff-79ef4a3dce58a9d303d49c5aee781e17R1484

Reply:
If you don't mind, let's try to keep this PR focussed on the RANDAO changes and take the next one once this one lands? (if possible ofc)

Reply:
Yes, of course. I just clarified why the code assumes having multiple transactions. Seems that needs to be fixed here. Hope @afck get well soon and fix that. He told he's a bit ill at the moment

Reply:
@afck So, I guess we should get rid of the `transactions` vector [here](https://github.com/poanetwork/parity-ethereum/blob/727fce39a1b8c922db27533b9d38bc879f707a32/ethcore/engines/authority-round/src/lib.rs#L1087-L1107) since there is only one transaction in this PR? Or add a comment that the vector can be needed for other transactions in the future?

Reply:
Done in https://github.com/paritytech/parity-ethereum/pull/10946/commits/0d7d0c8ecd13f658b9299fde19e9e15ebf1e788f.